src/transformers/models/cpmant/modeling_cpmant.py

```diff
@@ -0,0 +1,1090 @@
+            A sequence of tokens that is processed together as a unit.
+        span (`torch.Tensor` of shape `(batch_size, seq_len)`, *optional*):
+            A contiguous sequence of tokens within the input text.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    "The bare CPMAnt Model outputting raw hidden-states without any specific head on top.",
+    CPMANT_START_DOCSTRING,
+)
+class CPMAntModel(CPMAntPreTrainedModel):
+    def __init__(self, config: CPMAntConfig):
+        super().__init__(config)
+        self.encoder = CPMAntEncoder()
+        self.segment_embedding = nn.Embedding(config.segment_types, config.dim_model)
+        self.input_embedding = nn.Embedding(
+            config.vocab_size + config.prompt_types * config.prompt_length, config.dim_model
+        )
+        self.position_bias = CPMAntSegmentPositionEmbedding()
+        self.prompt_length = config.prompt_length
+
+    def get_input_embeddings(self):
+        embeddings = {
+            "segment": self.segment_embedding,
+            "input_ids": self.input_embedding,
+            "position": self.position_bias,
+        }
+        return embeddings
+
+    def set_input_embeddings(self, embeddings, **kwargs):
+        self.segment_embedding = embeddings["segment"]
+        self.input_embedding = embeddings["input_ids"]
+        self.position_bias = embeddings["position"]
+
+    def _prepare_attention_mask(self, input, span, context, length):
+        batch = input.size(0)
+        device = input.device
+        seqlen = input.size(1)
+        directional_mask_2d = torch.arange(seqlen, device=device) <= torch.arange(seqlen, device=device).view(-1, 1)
+        attention_mask = context[:, None, :] | (
+            context[:, :, None].logical_not() & directional_mask_2d.view(1, seqlen, seqlen)
+        )
+        attention_mask = attention_mask & (span[:, None, :] == span[:, :, None])
+        # mask for left paddding
+        mask_1d = torch.tensor(list(range(seqlen))[::-1], device=device)[None, :].repeat(batch, 1) < length[:, None]
+        attention_mask = mask_1d.view(batch, seqlen, 1) & mask_1d.view(batch, 1, seqlen) & attention_mask
+        return attention_mask
+
+    @add_start_docstrings_to_model_forward(CPMANT_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        processor_class=_TOKENIZER_FOR_DOC,
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=BaseModelOutput,
+        config_class=_CONFIG_FOR_DOC,
+    )
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        length: Optional[torch.Tensor] = None,
+        context: Optional[torch.Tensor] = None,
+        position: Optional[torch.Tensor] = None,
+        segment: Optional[torch.Tensor] = None,
+        span: Optional[torch.Tensor] = None,
+        return_dict: Optional[bool] = False,
+        **kwargs,
```

Comment:
keyword arguments that are needed for public models such as `output_attentions` seem to be missing here

Reply:
We added the `output_attentions `.